activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -93,6 +104,37 @@ def commit_records
         ite.each { |i| i.committed!(should_run_callbacks: false) }
       end
 
+      def run_commit_callbacks
+        while callback = commit_callbacks.shift
+          if @run_commit_callbacks
+            callback.call(self)
+          else
+            # if not running callbacks (ex: SavePoint transaction) move the callbacks to the parent transaction
+            connection.current_transaction.on_commit(&callback)
+          end
+        end
+      ensure
+        commit_callbacks.clear
+        rollback_callbacks.clear
```

Comment:
Since this is the same code as below in `run_rollback_callbacks` you could wrap `commit_callbacks.clear` and rollback_callbacks.clear` into a shared `clear_callbacks` method.